File: examples/fact/fact.c
#include <stdio.h>

int factorial(int n) {
    if (n == 0 || n == 1) {
        return 1;
    } else {
        return n * factorial(n - 1);
    }
}

int main() {
    int number;
    printf("Enter a number: ");
    scanf("%d", &number);

    if (number < 0) {
        printf("Invalid input. Please enter a non-negative integer.\n");
    } else {
        int result = factorial(number);
        printf("%d\n", result);
    }

    return 0;
}

File: examples/fact/fact.ex
defmodule Factorial do
  def of(n) do
    case n do
      0 -> 1
      1 -> 1
      n -> n * of(n - 1)
    end
  end
end

IO.puts("enter a number:")

number = case Integer.parse(IO.read(:stdio, :line)) do
    :error -> raise "Invalid input. Please enter a valid non-negative integer."
    {n, _} when n < 0 -> raise "Invalid input. Please enter a non-negative integer."
    {n, _} -> n
end

IO.puts(Factorial.of(number))
